activerecord/lib/active_record/railties/databases.rake

```diff
@@ -107,9 +110,12 @@ db_namespace = namespace :db do
     ActiveRecord::Tasks::DatabaseTasks.for_each(databases) do |name|
       # IMPORTANT: This task won't dump the schema if ActiveRecord::Base.dump_schema_after_migration is set to false
       task name do
-        if ActiveRecord::Base.dump_schema_after_migration
+        db_config = ActiveRecord::DatabaseConfigurations.new(databases).configs_for(env_name: Rails.env, name: name)
```

Comment:
This task has the same problem as above and needs to be updated in the same way.